### ISSUE TITLE
Indexing with torch tensors and NumPy arrays is different

### ISSUE THREAD

**Issue by happyalfred2016** [pytorch/pytorch#29522]
## 🐛 Bug

When I use a numpy.ndarray type boolean index for a tensor, it sometimes goes wrong when the tensor is a relatively small one.

## To Reproduce

```
np.random.seed(1024)
A = torch.tensor(np.random.rand(26,54))
W = (A > 0.5).numpy()
A[W]
```
Then it reports an error: "IndexError: too many indices for tensor of dimension 2"
However, if you transpose both A and W, it works fine! 
```
A = A.t()
W = W.T
C = A[W]
```
## Expected behavior

It should return a 1d tensor consisting of elements whose corresponding boolean index is True, like what it is done for the transposed tensor.

## Environment
Collecting environment information...
PyTorch version: 1.2.0
Is debug build: No
CUDA used to build PyTorch: 10.0
OS: Microsoft Windows 10 Pro
GCC version: (x86_64-posix-seh, Built by strawberryperl.com project) 8.3.0
CMake version: Could not collect
Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1050
Nvidia driver version: 431.70
cuDNN version: Could not collect
Versions of relevant libraries:
[pip] numpy==1.16.4
[pip] torch==1.2.0
[pip] torchvision==0.4.0
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.4                      245  
[conda] mkl-service               2.0.2            py36he774522_0  
[conda] mkl_fft                   1.0.14           py36h14836fe_0  
[conda] mkl_random                1.0.2            py36h343c172_0  
[conda] pytorch                   1.2.0           py3.6_cuda100_cudnn7_1    pytorch
[conda] torchvision               0.4.0                py36_cu100    pytorch

cc @mruberry

**Comment by ngimel** [comment]
It works if you don't cast index tensor to numpy, and directly use pytorch boolean tensor.

**Comment by happyalfred2016** [comment]
> It works if you don't cast index tensor to numpy, and directly use pytorch boolean tensor.

Yes, thanks for your reply. But I think there must be something wrong.

**Comment by mruberry** [comment]
Could be resolve with https://github.com/pytorch/pytorch/issues/11695 depending on the implementation.
